### ISSUE TITLE
[REVIEW]: Windrose

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#268]
**Submitting author:** @scls19fr (<a href="http://orcid.org/0000-0001-9987-4338">Sébastien Celles</a>)
**Repository:** <a href="https://github.com/python-windrose/windrose/" target ="_blank">https://github.com/python-windrose/windrose/</a>
**Version:** 1.6.5
**Editor:** @arfon
**Reviewer:** @amoeba
**Archive:** <a href="https://doi.org/10.5281/zenodo.1406384" target="_blank">10.5281/zenodo.1406384</a>

## Status

[![status](http://joss.theoj.org/papers/1b3fb0a7777ba65071e4d99b65a8c47d/status.svg)](http://joss.theoj.org/papers/1b3fb0a7777ba65071e4d99b65a8c47d)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/1b3fb0a7777ba65071e4d99b65a8c47d"><img src="http://joss.theoj.org/papers/1b3fb0a7777ba65071e4d99b65a8c47d/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/1b3fb0a7777ba65071e4d99b65a8c47d/status.svg)](http://joss.theoj.org/papers/1b3fb0a7777ba65071e4d99b65a8c47d)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/python-windrose/windrose/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

### Conflict of interest

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (such as being a major contributor to the software).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/python-windrose/windrose/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** Does the release version given match the GitHub release (1.6)?
- [x] **Authorship:** Has the submitting author (@scls19fr) made major contributions to the software?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** Have any performance claims of the software been confirmed?

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g. API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g. papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS. @tacaswell it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As as reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all JOSS reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by arfon** [comment]
Friendly reminder on this review @tacaswell 😁

**Comment by tacaswell** [comment]
- I am the lead developer of a dependency (Matplotlib), but I do not think that is a conflict of interest.
 - This is licensed under "CeCILL-B FREE SOFTWARE LICENSE AGREEMENT" which is labeled at the bottom as v1.0.  There is a CeCILL license on the OSI list is "Cea Cnrs Inria Logiciel Libre License, version 2.1 (CECILL-2.1)"
 - there is not a 'github release' but the version on pypi matches 1.6
 - this is a fork / continuation of an existing project.  The original author is clearly cited and included in the paper, have they been consulted?
 - are there docs other than the README?  If so I could not easily find them.
 - paper has no references, are any required?

**Comment by scls19fr** [comment]
> there is not a 'github release' but the version on pypi matches 1.6

Should we add a tag on GitHub? If that's necessary we can do it.

> - this is a fork / continuation of an existing project. The original author is clearly cited and included in the paper, have they been consulted?

@LionelR have been consulted see https://github.com/python-windrose/windrose/issues/54

> are there docs other than the README? If so I could not easily find them.

No other docs than the README

> paper has no references, are any required?

I don't really know what should be part of this field (I've never published such paper).

**Comment by tacaswell** [comment]
Yes, pushing tags that match at least the most recent version is (I think) required.  The tags are what are archived, give you a nice single-file source download, and you can use zenodo.org to get a DOI for the tag / release.

**Comment by scls19fr** [comment]
I've just release version 1.6.1 (Zenodo DOI https://zenodo.org/record/836356#.WX2aPdOLRZ0 , tag on GitHub, upload on PyPi https://pypi.python.org/pypi?%3Aaction=pkg_edit&name=windrose )

My release procedure is defined at https://github.com/python-windrose/windrose/issues/59

**Comment by tacaswell** [comment]
@arfon Who should be consulted about the license issue?

**Comment by arfon** [comment]
@tacaswell - it seems like the authors are discussing how to resolve this in https://github.com/python-windrose/windrose/issues/65 ?

**Comment by scls19fr** [comment]
PyPi doesn't support CeCILL-B license in classifiers for now (see https://github.com/pypa/pypi-legacy/issues/679 ) so as a workaround I commented out License field in classifier but licence is (as previously) correctly set to CeCILL-B thanks to https://github.com/python-windrose/windrose/blob/master/setup.py#L35 and https://github.com/python-windrose/windrose/blob/master/windrose/version.py#L5 )
I can be seen on https://pypi.python.org/pypi?:action=display&name=windrose&version=1.6.2

**Comment by arfon** [comment]
Hi @scls19fr, quoting from https://github.com/python-windrose/windrose/issues/65#issuecomment-320047090: 

> If that's an issue for JOSS I will create an other release with dual license.

If possible, I think it might be easier to dual-license this project. We're not in a position to offer legal advice to authors but I would be more comfortable if there was a license file present in the repository that was definitively listed in the OSI approved licenses.

**Comment by scls19fr** [comment]
windrose is now published with dual licensing: `CeCILL-B OR BSD-3-Clause`

**Comment by arfon** [comment]
Thanks @scls19fr. @tacaswell - are you nearly done with your review? There are still a fair number of checkboxes unchecked above :-)

**Comment by tacaswell** [comment]
I am going to be out-of-contact for the next 2 weeks, sorry :disappointed: and have not finished.

**Comment by arfon** [comment]
Friendly reminder to revisit this when you get a chance @tacaswell :-)

**Comment by arfon** [comment]
> Friendly reminder to revisit this when you get a chance @tacaswell :-)

*bump*

**Comment by arfon** [comment]
Just sent an email to @tacaswell. If we don't hear back from him in the next couple of weeks @scls19fr I think we'll have to find another reviewer.

**Comment by arfon** [comment]
@whedon list reviewers

**Comment by whedon** [comment]
Here's the current list of reviewers: https://bit.ly/joss-reviewers

**Comment by arfon** [comment]
@scls19fr - would you mind taking a look at the reviewer list here https://bit.ly/joss-reviewers and seeing if you can identify any suitable reviewers?

**Comment by scls19fr** [comment]
Being confortable with Ecology, stats, viz, system and Python programming, amoeba can probably be a suitable reviewer.

**Comment by arfon** [comment]
👋 @amoeba - would you be willing to finish up this review for us?

**Comment by amoeba** [comment]
Hey @arfon yes I would. I can tackle it this weekend.

**Comment by arfon** [comment]
Wonderful, thanks so much @amoeba.

@whedon assign @amoeba as reviewer

**Comment by arfon** [comment]
@whedon assign @amoeba as reviewer

**Comment by whedon** [comment]
OK, the reviewer is @amoeba

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by arfon** [comment]
@scls19fr - please could you take a look at the guidelines for the [JOSS paper structure](http://joss.theoj.org/about#paper_structure) and add some more detail to your `paper.md` please?

**Comment by scls19fr** [comment]
I will do this monday morning

**Comment by arfon** [comment]
> I will do this monday morning

👍thanks.

**Comment by amoeba** [comment]
Hey @scls19fr I've posted my partial review at https://github.com/python-windrose/windrose/issues/77 while we wait on the updates you are making to the paper. Let me know if you have any questions over there.

**Comment by arfon** [comment]
:wave: @scls19fr - how are you getting along here?

**Comment by scls19fr** [comment]
I've just reply today to @amoeba in https://github.com/python-windrose/windrose/issues/77#issuecomment-385725732 to address most of his comments
Sorry for delay.

**Comment by amoeba** [comment]
Looking good! I made some comments [over there](https://github.com/python-windrose/windrose/issues/77#issuecomment-386989721) and still have a few remaining issues for the author to tackle.

**Comment by arfon** [comment]
@scls19fr - how are we getting along here?

**Comment by scls19fr** [comment]
This end of year is very difficult and this article is (unfortunately) not on top of my ToDo list. I will tackle this asap but I think you can tag it as "paused" for the next 3 weeks. Sorry and thanks again @amoeba for helping to improve quality.

**Comment by arfon** [comment]
:wave: @scls19fr - friendly reminder to revisit this sometime soon please.

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #268 with the following error: 

   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    11    0    11    0     0     59      0 --:--:-- --:--:-- --:--:--    60
Error reading bibliography ./paper.bib (line 89, column 7):
unexpected ","
expecting white space
Error running filter pandoc-citeproc:
Filter returned error status 1
Looks like we failed to compile the PDF

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
Shortening title

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
LGTM but let's wait @amoeba opinion.

**Comment by amoeba** [comment]
Content looks good, thanks @scls19fr. I'm not sure if this was happening before but the PDF version of the paper doesn't show the figures in-line so the examples section is confusing. Could you either find a solution to that or change the bulleted list to a paragraph of text with parenthetical figure references? I'm about to be out of range through Monday in the USA so apologies in advance if my response time drops to zero for a few days here.

**Comment by scls19fr** [comment]
This change have been done by a contributor in https://github.com/python-windrose/windrose/pull/100
Will revert it. But it adds `-`.

**Comment by scls19fr** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00268/joss.00268/10.21105.joss.00268.pdf)

**Comment by scls19fr** [comment]
Not sure if we can include figure at a specific position without seeing this `-`.

**Comment by amoeba** [comment]
LGTM, thanks!

@arfon: Review is complete and I approve the submission assuming the version discrepancy (1.6 vs latest) be resolved at the time of archive.

Thanks for listening to my feedback and making changes in a timely manner @scls19fr!

**Comment by amoeba** [comment]
> Not sure if we can include figure at a specific position without seeing this -.

Oh wow, I didn't even notice that and thought it was just part of the images. I'm not super worried about it at this point.

**Comment by arfon** [comment]
@scls19fr - At this point could you make an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive? I can then move forward with accepting the submission.

**Comment by scls19fr** [comment]
v1.6.5 published

See:
- https://zenodo.org/account/settings/github/repository/python-windrose/windrose# 
- https://pypi.org/project/windrose/
- https://github.com/python-windrose/windrose

However I can't edit first thread from

    Version: 1.6

to

    Version: 1.6.5

**Comment by arfon** [comment]
> Version: 1.6.5

Done.

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.1406384 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.1406384" target="_blank">10.5281/zenodo.1406384</a> is the archive.

**Comment by scls19fr** [comment]
Works fine now... sorry for noise

**Comment by arfon** [comment]
@amoeba @tacaswell - many thanks for your reviews here ✨ 

@scls19fr - your paper is now accepted into JOSS and your DOI is https://doi.org/10.21105/joss.00268 :zap: :rocket: :boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.00268/status.svg)](https://doi.org/10.21105/joss.00268)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.00268">
  <img src="http://joss.theoj.org/papers/10.21105/joss.00268/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.00268/status.svg
   :target: https://doi.org/10.21105/joss.00268
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.00268/status.svg)](https://doi.org/10.21105/joss.00268)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3Nzk=

**Comment by scls19fr** [comment]
Thanks @amoeba @tacaswell @arfon ... and @whedon
